Rb.Forms.Barcode.Droid/BarcodeScannerRenderer.cs

```diff
@@ -73,12 +83,20 @@ public static void Init(Configuration config)
                 return;
             }
 
-            await Task.Run(() => CameraService.OpenCamera(holder));
-        }
+            var windowManager = Context.GetSystemService(Android.Content.Context.WindowService).JavaCast<IWindowManager>();
+            var rotation = windowManager.DefaultDisplay.Rotation;
 
-        public void SurfaceChanged(ISurfaceHolder holder, global::Android.Graphics.Format format, int width, int height)
-        {
-            this.Debug("SurfaceChanged");
+            // landscape mode
+            if ((int) rotation % 2 == 0) {
+                CameraService.SetViewSize(height, width);
+            } else {
+                CameraService.SetViewSize(width, height);
+            }
+
```

Comment:
Could be clearer here so a comment isn't needed:

```cs
  var surfaceOrientation = WindowManager.DefaultDisplay.Rotation;
 >>var surfaceOrientation = WindowManager.DefaultDisplay.Rotation;
  if (surfaceOrientation == SurfaceOrientation.Rotation0 ||  
                  surfaceOrientation == SurfaceOrientation.Rotation180) 
  {
          ... 
  }
      
but I don't think the "natural rotation" is the best way to determine whether it's landscape or portrait, because natural rotation means different things for different device profiles.   Plus, isn't `rotation % 0` == 0 for when a phone that's taller than it is wide when that phone is in portrait mode?  

Just an idea, but maybe the `width` and `height` arguments could be used to determine directly portrait or landscape, or the CameraService could have `Width` and `Height` properties to determine this itself.